### PR TITLE
feat: add skip blur option

### PR DESCRIPTION
**Pull request by mrpmohiburrahman**
### Summary

Resolves: https://github.com/callstack/react-native-testing-library/issues/1677

### Test plan

All tests pass

### CODE REVIEW

File: src/user-event/type/type.ts [hunk 1]
```diff
@@ -69,7 +70,9 @@ export async function type(
 
   dispatchEvent(element, 'endEditing', EventBuilder.TextInput.endEditing(finalText));
 
-  dispatchEvent(element, 'blur', EventBuilder.Common.blur());
+  if (!options?.skipBlur) {
```
Comment (mdjastrzebski) [inline src/user-event/type/type.ts:73]: skip blur option should disable both endEditing and blur together, as in in native platforms (iOS, android) they always happen together.

**Review by mdjastrzebski — CHANGES_REQUESTED** [review]
I've left comment about events affected by this flag. Please add also documentation entry for the option.

File: src/user-event/type/type.ts (hunk 1 above)
Comment (mrpmohiburrahman) [reply src/user-event/type/type.ts:71]: Made the changes on the latest commit.

**Comment by mdjastrzebski** [comment]
Document to update: https://github.com/callstack/react-native-testing-library/blob/main/website/docs/12.x/docs/api/events/user-event.mdx.

Please also add units test(s) for behavior with `skipBlur` option set.

**Comment by mrpmohiburrahman** [comment]
> Document to update: https://github.com/callstack/react-native-testing-library/blob/main/website/docs/12.x/docs/api/events/user-event.mdx.
> 
> Please also add units test(s) for behavior with `skipBlur` option set.

Hi, @mdjastrzebski , 

I have updated doc and added unit test.
please, check.
Also note that, I am very new to this.

**Review by mdjastrzebski — APPROVED** [review]
Ready to merge! @mrpmohiburrahman thank you for this contribution.

**Pull request merged**